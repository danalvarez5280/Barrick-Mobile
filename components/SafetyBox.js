import React, { Component } from 'react';
import {
  AppRegistry,
  Button,
  Image,
  Keyboard,
  Slider,
  StatusBar,
  StyleSheet,
  Text,
  TextInput,
  TouchableHighlight,
  View,
} from 'react-native';

import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';

import safetyChecked from './assets/checked.png';
import notChecked from './assets/not-checked.png';

//colors
// gold #C4900F
// blue #122732
//off white #CED5E0
// off blue #324A5E
class SafetyBox extends Component {
  constructor(props) {
    super();
    this.state = {
      checked: props.checked,
      list: props.list,
      check: props.safeCheck,
    }
    this.changeColor = this.changeColor.bind(this);
  };

  changeColor() {
    console.log('hi dan');
    this.state.checked ?
    this.setState({
      checked: false,
    }) :
    this.setState({
      checked: true,
    })
  };


  render() {
    let boxChecked = this.state.checked ? safetyChecked : notChecked;
    let boxStyle = this.state.checked ? styles.gold : styles.blue;
    return(
      <TouchableHighlight onPress={ () => this.changeColor() }>
        <View style={ boxStyle }>
          <Image source={ boxChecked } style={ styles.checkBox } />
          <Text style={ styles.white }>{ this.state.check }</Text>
        </View>
      </TouchableHighlight>
    )
  }
};

const styles = StyleSheet.create({
  white: {
    color: 'white',
    fontSize: 12,
    fontWeight: 'bold',
    marginLeft: 20,
  },
  blue: {
    alignItems: 'center',
    backgroundColor: '#122732',
    borderRadius: 15,
    display: 'flex',
    flexDirection: 'row',
    height: 40,
    marginTop: 10,
    padding: 5,
    width: 220,
  },
  gold: {
    alignItems: 'center',
    backgroundColor: '#C4900F',
    borderRadius: 15,
    display: 'flex',
    flexDirection: 'row',
    height: 40,
    marginTop: 10,
    padding: 5,
    width: 220,
  },
  checkBox: {
    height: 24,
    width: 24,
  }
});

export default SafetyBox;
