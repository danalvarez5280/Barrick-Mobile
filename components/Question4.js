import React, { Component } from 'react';
import {
  AppRegistry,
  Button,
  Image,
  Keyboard,
  Slider,
  StatusBar,
  StyleSheet,
  Text,
  TextInput,
  TouchableHighlight,
  View,
} from 'react-native';

import CheckBox from 'react-native-checkbox';

import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';

import SafetyBox from './SafetyBox';

//colors
// gold #C4900F
// blue #122732
//off white #CED5E0
// off blue #324A5E
class Question4 extends Component {
  constructor(props) {
    super();
    console.log('props', props);
    this.state = {
      checkedControls: props.info.checkedControls,
      specificSafetyMeasures: props.info.specificSafetyMeasures,
      checkList: props.info.checkList,
    }
    this.goBack = this.goBack.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  };

  controlsChecked() {
    this.props.controlsChecked()
  };

  goBack() {
    this.props.goBack(3);
  };

  handleSubmit() {
    this.props.submitLogin(this.state);
  };

  noInfo() {
    Alert.alert('You need to input a CHECK ALL SAFETY MEASURES')
  };

  // safetyCheck() {
  //   let checks = this.state.checkList;
  //   const safetyChecks = checks.map((check, index) =>
  //     if(check === (<SafetyBox key={ index } safeCheck={ check } checked={ true } />)) {
  //       console.log('checks passed');
  //       controlsChecked()
  //     }
  //   );
  // 
  // }

  render() {
    console.log('props4', this.state.checkList);
    let checks = this.state.checkList;
    const safetyChecks = checks.map((check, index) =>
      <SafetyBox
        key={ index }
        safeCheck={ check }
        checked={ false } />
    );

    return(
      <KeyboardAwareScrollView
        scrollEnabled={ true }
        contentContainerStyle={ styles.display }>
        <View style={ styles.question }>
          <Text style={ styles.prompt }>Safety Measure Checklist</Text>
          <View style={ styles.list }>
          { safetyChecks }
          </View>
          <Text style={ styles.prompt }>Specific Concerns</Text>
          <TextInput
            style={ styles.input }
            multiline= { true }
            placeholder='Specific Concerns'
            onChangeText={ (specificSafetyMeasures) => this.setState({ specificSafetyMeasures })}
            value={ this.state.specificSafetyMeasures } />
          <Button onPress={ this.handleSubmit } title='SUBMIT'/>
          <Button onPress={ this.goBack } title='BACK'/>
        </View>
      </KeyboardAwareScrollView>
    )
  }
};
const styles = StyleSheet.create({
  display: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    marginTop: 10,
    padding: 0,
    height: '100%',
    width: '100%',
  },
  input: {
    backgroundColor: '#CED5E0',
    color: '#122732',
    fontSize: 20,
    height: 100,
    marginBottom: 10,
    padding: 5,
    width: 250,
  },
  list: {
    alignItems: 'flex-start',
    display: 'flex',
    flexDirection: 'column',
    margin: 10,
  },
  question: {
    alignItems: 'center',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    marginTop: 0,
  },
  prompt: {
    backgroundColor: '#C4900F',
    color: '#122732',
    fontWeight: 'bold',
    padding: 5,
    textAlign: 'center',
    width: 250,
  },
  riskIcon: {
    height: 80,
    margin: 15,
    width: 80,
  },
  slider: {
    width: 200
  }
});

export default Question4;
