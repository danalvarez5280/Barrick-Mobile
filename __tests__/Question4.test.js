import 'react-native';
import React from 'react';
import { shallow, mount } from 'enzyme';
import Question4 from '../components/Question4';
import { Button, Image, Slider, Text, TextInput, TouchableHighlight } from 'react-native';

import SafetyBox from '../components/SafetyBox';

import renderer from 'react-test-renderer';

const q4Submit = jest.fn();
const controlsChecked = jest.fn();
const logOut = jest.fn();
const goBack = jest.fn();
const mockState = {
  checkedControls: true,
  specificSafetyMeasures: 'I could stub my toe',
  checkList: ['wear rubber gloves', 'lockout/tagout', 'check locks']
};

const setup = () => {
  return shallow(<Question4
    submitLogin={ q4Submit }
    controlsChecked={ controlsChecked }
    goBack={ goBack }
    info={ mockState }
    logOut={ logOut } />)
};



it('renders correctly', () => {

  const tree = renderer.create(
    <Question4
    submitLogin={ q4Submit }
    goBack={ goBack }
    controlsChecked={ controlsChecked }
    info={ mockState }
    logOut={ logOut } />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
