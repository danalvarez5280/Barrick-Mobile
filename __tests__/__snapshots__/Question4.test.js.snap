// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "height": 550,
      "justifyContent": "space-between",
      "overflow": "scroll",
      "padding": 0,
      "width": "100%",
    }
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutoAutomaticScroll={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  onScroll={[Function]}
  scrollEnabled={true}
  scrollEventThrottle={0}
  showsVerticalScrollIndicator={true}
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
          "marginTop": 0,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "#C4900F",
            "color": "#122732",
            "fontWeight": "bold",
            "padding": 5,
            "textAlign": "center",
            "width": 250,
          }
        }
      >
        Safety Measure Checklist
      </Text>
      <RCTScrollView
        contentContainerStyle={
          Object {
            "alignItems": "flex-start",
            "display": "flex",
            "flexDirection": "column",
            "margin": 10,
            "overflow": "scroll",
          }
        }
      >
        <View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#122732",
                  "borderRadius": 15,
                  "display": "flex",
                  "flexDirection": "row",
                  "height": 40,
                  "marginTop": 10,
                  "padding": 5,
                  "width": 225,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../components/assets/checked.png",
                  }
                }
                style={
                  Object {
                    "height": 24,
                    "width": 24,
                  }
                }
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "fontSize": 12,
                    "fontWeight": "bold",
                    "marginLeft": 20,
                  }
                }
              >
                wear rubber gloves
              </Text>
            </View>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#122732",
                  "borderRadius": 15,
                  "display": "flex",
                  "flexDirection": "row",
                  "height": 40,
                  "marginTop": 10,
                  "padding": 5,
                  "width": 225,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../components/assets/checked.png",
                  }
                }
                style={
                  Object {
                    "height": 24,
                    "width": 24,
                  }
                }
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "fontSize": 12,
                    "fontWeight": "bold",
                    "marginLeft": 20,
                  }
                }
              >
                lockout/tagout
              </Text>
            </View>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#122732",
                  "borderRadius": 15,
                  "display": "flex",
                  "flexDirection": "row",
                  "height": 40,
                  "marginTop": 10,
                  "padding": 5,
                  "width": 225,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../components/assets/checked.png",
                  }
                }
                style={
                  Object {
                    "height": 24,
                    "width": 24,
                  }
                }
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "fontSize": 12,
                    "fontWeight": "bold",
                    "marginLeft": 20,
                  }
                }
              >
                check locks
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "#C4900F",
            "color": "#122732",
            "fontWeight": "bold",
            "padding": 5,
            "textAlign": "center",
            "width": 250,
          }
        }
      >
        Specific Safety Measures
      </Text>
      <TextInput
        allowFontScaling={true}
        multiline={true}
        onChangeText={[Function]}
        placeholder="Specific Concerns"
        style={
          Object {
            "backgroundColor": "#CED5E0",
            "color": "#122732",
            "fontSize": 20,
            "height": 100,
            "marginBottom": 10,
            "padding": 5,
            "width": 250,
          }
        }
        value="I could stub my toe"
      />
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={undefined}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            SUBMIT
          </Text>
        </View>
      </View>
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={undefined}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            BACK
          </Text>
        </View>
      </View>
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={undefined}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            LOG OUT
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
